velox/expression/Expr.cpp

```diff
@@ -1207,8 +1262,26 @@ void Expr::evalAll(
   LocalSelectivityVector mutableRemainingRowsHolder(context);
 
   inputValues_.resize(inputs_.size());
+  EvalCtx::ErrorVectorPtr argumentErrors;
+  EvalCtx::ErrorVectorPtr errors;
+  if (context.errors()) {
+    context.swapErrors(errors);
+  }
+
   for (int32_t i = 0; i < inputs_.size(); ++i) {
+    ScopedVarSetter throwErrors(
```

Comment:
why is this in the loop and not in a block outside. i mean we are setting and resetting same variable.

maybe
{
 throwArgumentErrors= ..
 VarSetter ..;
 // loop.
}